hudi-common/src/main/java/org/apache/hudi/avro/HoodieAvroUtils.java

```diff
@@ -292,53 +292,53 @@ public static GenericRecord stitchRecords(GenericRecord left, GenericRecord righ
     return result;
   }
 
-  /**
-   * Given a avro record with a given schema, rewrites it into the new schema while setting fields only from the old
-   * schema.
-   */
-  public static GenericRecord rewriteRecord(GenericRecord record, Schema newSchema) {
-    return rewrite(record, getCombinedFieldsToWrite(record.getSchema(), newSchema), newSchema);
-  }
-
   /**
    * Given a avro record with a given schema, rewrites it into the new schema while setting fields only from the new
    * schema.
+   * NOTE: Here, the assumption is that you cannot go from an evolved schema (schema with (N) fields)
+   * to an older schema (schema with (N-1) fields). All fields present in the older record schema MUST be present in the
+   * new schema and the default/existing values are carried over.
+   * This particular method does the following things :
+   * a) Create a new empty GenericRecord with the new schema.
+   * b) For GenericRecord, copy over the data from the old schema to the new schema or set default values for all fields of this
+   * transformed schema
+   * c) For SpecificRecord, hoodie_metadata_fields have a special treatment. This is done because for code generated
+   * AVRO classes (HoodieMetadataRecord), the avro record is a SpecificBaseRecord type instead of a GenericRecord.
```

Comment:
There is no such class as HoodieMetadataRecord. Am I missing something here?

Reply:
AVRO classes -> avro classes? Let us have a uniform way of writing "avro". :)

Reply:
There are many places with different caps naming, it's impossible to control that, but fixed it anyways :) For the second comment, please read my comment, `HoodieMetadataRecord` is a code generated class.